### PR TITLE
etcdutil, leadership: avoid redundant created watch channel

### LINKED ISSUE #7351: tso server keep recreating watch channel after api server restart
### What did you do?

Restart api server during tso server running.

### What did you expect to see?

tso server recover after api server restart.

### What did you see instead?

tso server flushing logs in all tso servers.

```
[2023/11/10 15:37:09.725 +08:00] [INFO] [etcdutil.go:718] ["watch channel is created in watch loop"] [revision=87752] [name=keyspace-watcher] [key=/pd/7298529810357898783/tso/keyspace_groups/membership/00000]
[2023/11/10 15:37:10.724 +08:00] [WARN] [etcdutil.go:743] ["watch channel is blocked for a long time, recreating a new one in watch loop"] [timeout=8m8.998628736s] [revision=87752] [name=keyspace-watcher] [key=/pd/7298529810357898783/tso/keyspace_groups/membership/00000]
[2023/11/10 15:37:10.725 +08:00] [INFO] [etcdutil.go:718] ["watch channel is created in watch loop"] [revision=87752] [name=keyspace-watcher] [key=/pd/7298529810357898783/tso/keyspace_groups/membership/00000]
[2023/11/10 15:37:11.724 +08:00] [WARN] [etcdutil.go:743] ["watch channel is blocked for a long time, recreating a new one in watch loop"] [timeout=8m9.998671435s] [revision=87752] [name=keyspace-watcher] [key=/pd/7298529810357898783/tso/keyspace_groups/membership/00000]
```

### What version of PD are you using (`pd-server -V`)?

```
Release Version: v7.1.1-serverless
Edition: Community
Git Commit Hash: d39c79fdb49151f010e73d2c36832a6acae6bd70
```

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Close #7351

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by rleungx — APPROVED** [review]

**Review by CabinfeverB — APPROVED** [review]

File: pkg/utils/etcdutil/etcdutil.go [hunk 1]
```diff
@@ -707,7 +707,6 @@ func (lw *LoopWatcher) watch(ctx context.Context, revision int64) (nextRevision
 	}()
 	ticker := time.NewTicker(RequestProgressInterval)
 	defer ticker.Stop()
-	lastReceivedResponseTime := time.Now()
```
Comment (CabinfeverB) [inline pkg/utils/etcdutil/etcdutil.go:710]: Don't need to delete the same line in `leadership.go`?

Comment (CabinfeverB) [reply pkg/utils/etcdutil/etcdutil.go:710]: I got it, API server never meets this problem.

Comment (lhy1024) [reply pkg/utils/etcdutil/etcdutil.go:710]: No, `leadership` has other healthy check `if time.Since(lastReceivedResponseTime) > unhealthyTimeout `

**Pull request merged**